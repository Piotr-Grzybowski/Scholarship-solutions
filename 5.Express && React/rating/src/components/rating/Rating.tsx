import Score from "../score/Score";
import { rating } from "../types";
import ListItem from "@mui/material/ListItem";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";

function Rating({ name, score, content }: rating) {
  return (
    <ListItem
      sx={{ display: "flex", justifyContent: "space-evenly" }}
      data-testid="rating"
    >
      <ListItemIcon sx={{ marginRight: "10px", maxWidth: "25%" }}>
        <Score score={score}></Score>
      </ListItemIcon>
      <ListItemText
        primary={name}
        sx={{ marginRight: "5%", maxWidth: "10%" }}
      />
      <ListItemText
        primary={content.substring(0, 120) + "..."}
        sx={{ maxWidth: "60%", textAlign: "justify" }}
      />
    </ListItem>
  );
}

export default Rating;
