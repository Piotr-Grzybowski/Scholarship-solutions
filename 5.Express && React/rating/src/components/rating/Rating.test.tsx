import Rating from "./Rating";
import renderer from "react-test-renderer";

describe("Testing AverageScore", () => {
  const data = {
    recordId: "1",
    name: "John",
    score: 3,
    content: "Nice place to take a quick sip of tea",
  };

  test("when given result Rating should render with proper data", () => {
    const component = renderer.create(<Rating {...data}></Rating>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
