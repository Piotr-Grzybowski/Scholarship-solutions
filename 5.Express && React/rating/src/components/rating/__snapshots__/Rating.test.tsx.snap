// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AverageScore when given result Rating should render with proper data 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-q5n5zf-MuiListItem-root"
  data-testid="rating"
  disabled={false}
>
  <div
    className="MuiListItemIcon-root css-18jj6cy-MuiListItemIcon-root"
  >
    <div
      data-testid="score"
    >
      <svg
        data-testid="yellow-star"
      >
        yellowStar.svg
      </svg>
      <svg
        data-testid="yellow-star"
      >
        yellowStar.svg
      </svg>
      <svg
        data-testid="yellow-star"
      >
        yellowStar.svg
      </svg>
      <svg
        data-testid="gray-star"
      >
        grayStar.svg
      </svg>
      <svg
        data-testid="gray-star"
      >
        grayStar.svg
      </svg>
    </div>
  </div>
  <div
    className="MuiListItemText-root css-1owdhls-MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
    >
      John
    </span>
  </div>
  <div
    className="MuiListItemText-root css-16jj254-MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
    >
      Nice place to take a quick sip of tea...
    </span>
  </div>
</li>
`;
